packages/ui/src/themes/buildTheme.ts

```diff
@@ -0,0 +1,33 @@
+import type { PartialTheme, ParsedElements } from '~/contexts/AppearanceContext';
+
+/**
+ * Given a complete theme object minus descriptors, returns a full ParsedElements object with generated descriptors.
+ */
+export function buildTheme(p: PartialTheme): ParsedElements {
+  const theme: Partial<ParsedElements> = {};
+
+  for (const descriptor in p) {
+    if (p[descriptor as keyof PartialTheme]) {
+      const { className, style } = p[descriptor as keyof PartialTheme]!;
+      theme[descriptor as keyof ParsedElements] = {
+        descriptor: `cl-${descriptor}`,
+        className: className ?? 'debug',
+        style: style ?? {},
+      };
+    }
+  }
+
+  return theme as ParsedElements;
+}
+
+export function mergeTheme(a: ParsedElements, b: ParsedElements): ParsedElements {
+  const mergedTheme = { ...a };
+
+  for (const d in mergedTheme) {
+    const descriptor = d as keyof ParsedElements;
+    mergedTheme[descriptor].className = [mergedTheme[descriptor].className, b[descriptor].className].join(' ');
```

Comment:
Should we account for empty here? Or is it happening elsewhere?
```suggestion
    mergedTheme[descriptor].className = [mergedTheme[descriptor].className, b[descriptor].className].filter(Boolean).join(' ');
```

Reply:
yeah this is another place where we're assuming we're provided with two complete themes. When we remove that constraint we will need to add logic to handle situations where a descriptor is in a but not in b, and vice versa.